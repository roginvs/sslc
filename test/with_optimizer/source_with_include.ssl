#include <say_hello.h>

variable test_suite_errors := 0;
variable test_suite_verbose := true;


#ifndef assertEquals
    procedure assertEquals(variable desc, variable a, variable b) begin
       if (a != b or typeof(a) != typeof(b)) then begin
          display_msg("Assertion failed \""+desc+"\": "+a+" != "+b);
          test_suite_errors ++;
       end else if (test_suite_verbose) then begin
          display_msg("Assert \""+desc+"\" ok");
       end
    end
    
    procedure assertNotEquals(variable desc, variable a, variable b) begin
       if (a == b) then begin
          display_msg("Assertion failed \""+desc+"\": "+a+" == "+b);
          test_suite_errors ++;
       end else if (test_suite_verbose) then begin
          display_msg("Assert \""+desc+"\" ok");
       end
    end
#endif

procedure some_tests begin
    variable var1;
    var1 = "test1";

    test_suite_errors := 0;
    display_msg("Testing something...");
    call assertEquals("test1", var1, true);

    display_msg("All done");
end


procedure map_enter_p_proc begin
    call sayHello("Main file");
    display_msg("Starting tests");
    call some_tests();
 end