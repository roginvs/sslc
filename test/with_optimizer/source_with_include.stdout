Set include directory: ../include
Compiling source_with_include.ssl
[Warning] <source_with_include.ssl>:46: End of input with no newline, supplemented newline
     
[Message] <source_with_include.ssl>:17:75: Eliminating procedure assertNotEquals
[Message] <source_with_include.ssl>:12:40: Constant propagating global variable test_suite_verbose
[Message] <none>:-1: Eliminating unused global variable test_suite_verbose
[Message] <source_with_include.ssl>:12:51: Eliminating constant if expression (true)
[Message] <none>:-1: Removing unused string 'assertNotEquals' from program namespace
[Message] <none>:-1: Removing unused string 'test_suite_verbose' from program namespace
[Message] <none>:-1: Removing unused string ' == ' from program stringspace
